src/components/Company/PaySchedule/PaySchedule.tsx

```diff
@@ -0,0 +1,361 @@
+  companyId: string
+  defaultValues?: PayScheduleDefaultValues
+}
+
+export const PaySchedule = ({
+  companyId,
+  defaultValues,
+  ...props
+}: PayScheduleProps & BaseComponentInterface) => {
+  useI18n('Company.PaySchedule')
+  return (
+    <BaseComponent {...props}>
+      <Root companyId={companyId} defaultValues={defaultValues}>
+        {props.children}
+      </Root>
+    </BaseComponent>
+  )
+}
+
+const Root = ({ companyId, children, defaultValues }: PayScheduleProps) => {
+  const { baseSubmitHandler, onEvent } = useBase()
+  const [mode, setMode] = useState<MODE>('LIST_PAY_SCHEDULES')
+  const [currentPaySchedule, setCurrentPaySchedule] = useState<
+    (Schemas['Pay-Schedule'] & { version?: string }) | null
+  >(null)
+  const transformedDefaultValues: PayScheduleInputs = useMemo(() => {
+    return {
+      frequency: defaultValues?.frequency ?? 'Every week',
+      anchor_pay_date: defaultValues?.anchor_pay_date
+        ? parseDate(defaultValues.anchor_pay_date)
+        : undefined,
+      anchor_end_of_pay_period: defaultValues?.anchor_end_of_pay_period
+        ? parseDate(defaultValues.anchor_end_of_pay_period)
+        : undefined,
+      day_1: defaultValues?.day_1 ?? undefined,
+      day_2: defaultValues?.day_2 ?? undefined,
+      custom_name: defaultValues?.custom_name ?? '',
+      auto_pilot: defaultValues?.auto_pilot?.toString() ?? '',
+      custom_twice_per_month: 'false',
+    }
+  }, [defaultValues])
+
+  const [payScheduleDraft, setPayScheduleDraft] = useState<PaySchedulePreviewDraft | null>(null)
+  const {
+    data: payPreviewData,
+    error: paySchedulePreviewError,
+    isLoading,
+  } = useGetPaySchedulePreview(
+    companyId,
+    payScheduleDraft as PaySchedulePreviewDraft, // Casting to non-null because we know it's not null from the enabled prop
+    !!payScheduleDraft,
+  )
+
+  const { data: paySchedules } = useGetAllPaySchedules(companyId)
+
+  const createPayScheduleMutation = useCreatePaySchedule()
+  const updatePayScheduleMutation = useUpdatePaySchedule()
+
+  const formMethods = useForm<PayScheduleInputs, unknown, PayScheduleOutputs>({
+    resolver: valibotResolver(PayScheduleSchema),
+    defaultValues: transformedDefaultValues,
+  })
+
+  const allValues = formMethods.watch()
+
+  // Set the custom_twice_per_month value based on the frequency and day_1 and day_2 values as it is not set by the API call
+  useEffect(() => {
+    if (
+      allValues.frequency === 'Twice per month' &&
+      allValues.day_1 === 15 &&
+      allValues.day_2 === 31 &&
+      allValues.custom_twice_per_month === undefined
+    ) {
+      formMethods.setValue('custom_twice_per_month', `false`)
+    } else if (
+      allValues.frequency === 'Twice per month' &&
+      allValues.custom_twice_per_month === undefined
+    ) {
+      formMethods.setValue('custom_twice_per_month', `true`)
+    }
+  }, [
+    allValues.frequency,
+    allValues.day_1,
+    allValues.day_2,
+    formMethods,
```

Comment:
I ran into an infinite loop resulting from use of `formMethods` in the dep array 😅 see slack message from a bit ago here (before your time) https://gustohq.slack.com/archives/C071WFJUNF5/p1734377019109969

The fix is pretty straightforward, you can just destructure `setValue` from `formMethods` and include that in the dep array instead which should work the same!

Reply:
Resolved